model/gen-ai/registry.yaml

```diff
@@ -253,6 +255,155 @@ groups:
+                    "filename": "correlation_matrix.png",
+                    "mime_type": "image/png"
+                  }
+                },
+                {
+                  "type": "file",
+                  "file": {
+                    "file_id": "file-data001",
+                    "filename": "analysis_summary.pdf",
+                    "path": "sandbox:/mnt/data/analysis_summary.pdf",
+                    "mime_type": "application/pdf"
+                  }
+                }
+              ],
+              "execution_time_ms": 3427
+            }
+
+      - id: gen_ai.tool.definitions
+        stability: development
+        type: any
+        brief: The list of tool definitions available to the GenAI agent or model.
+        note: >
+          The value of this attribute matches provider-specific tool definition
+          format.
+          It's expected to be an object - in case a serialized string is available
+          to the instrumentation, the instrumentation SHOULD do the best effort to
+          deserialize it to an object.
+
+          Since this attribute could be large, it's NOT RECOMMENDED to populate
+          it by default. Instrumentations MAY provide a way to enable
+          populating this attribute.
+        examples: 
+        - |
+            [
+              {
+                "type": "function",
+                "name": "get_current_weather",
+                "description": "Get the current weather in a given location",
+                "parameters": {
+                  "type": "object",
+                  "properties": {
+                    "location": {
+                      "type": "string",
+                      "description": "The city and state, e.g. San Francisco, CA"
+                    },
+                    "unit": {
+                      "type": "string",
+                      "enum": [
+                        "celsius",
+                        "fahrenheit"
+                      ]
+                    }
+                  },
+                  "required": [
+                    "location",
+                    "unit"
+                  ]
+                }
+              }
+            ]
+      # Add these new attributes to the registry.gen_ai group in registry.yaml
+
+      - id: gen_ai.orchestrator.agent_definitions 
+        stability: development
+        type: any
+        brief: Definitions of agents available to the orchestrator for multi-agent workflows.
+        note: |
+          This attribute contains an array of objects representing agent definitions that are available to the orchestrator for potential use in multi-agent workflows.
+          
+          The structure is an array where each object contains:
+          
+          - 'agent_id': Unique identifier for the agent
+          - 'name': Human-readable name of the agent
+          - 'role': Classification of the agent (e.g., "specialist", "coordinator", "validator")
```

Comment:
One of the feedback I received in my now closed PR (#2551), was that name is sufficient and the role can be inferred from the name itself. As most users never do anything custom, less attributes the better IMO.